cores/esp8266/uart.c

```diff
@@ -156,20 +151,33 @@ uart_peek_char_unsafe(uart_t* uart)
    
     //without the following if statement and body, there is a good chance of a fifo overrun
     if (uart_rx_buffer_available_unsafe(uart->rx_buffer) == 0)
+        // hw fifo can't be peeked, data need to be copied to sw
         uart_rx_copy_fifo_to_buffer_unsafe(uart);
     
     return uart->rx_buffer->buffer[uart->rx_buffer->rpos];
 }
 
-inline int 
+// taking data straight from hw fifo: loopback-test BW jumps by 19%
+inline int
```

Comment:
This new code makes sense to me.
However, the code duplication here is increasing. Could you please make an inline function for reading one byte from the buffer, and one inline function for advancing the buffer pos? It should take as argument uart->rx_buffer.